pep-0688.rst

```diff
@@ -0,0 +1,275 @@
+In stub files, ``types.Buffer`` may be used as a base class to
+indicate that a class implements the buffer protocol.
+
+For example, ``bytes`` may be declared as follows in a stub:
+
+.. code-block:: python
+
+   class bytes(types.Buffer, Sequence[int]):
+       def decode(self, ...): ...
+       ...
+
+The ``types.Buffer`` class does not require any special treatment
+in type checkers.
+
+Equivalent for older Python versions
+------------------------------------
+
+New typing features are usually backported to older Python versions
+in the `typing_extensions <https://pypi.org/project/typing-extensions/>`_
+package. Because the buffer protocol
+is accessible only in C, ``types.Buffer`` cannot be implemented
+in a pure-Python package like ``typing_extensions``. As a temporary
+workaround, a ``typing_extensions.Buffer``
+`Abstract Base Class <Buffer ABC_>`__ will be provided on Python versions
+that do not have ``types.Buffer`` available. For the benefit of
+static type checkers, ``typing_extensions.Buffer`` can be used as
+a base class in stubs to mark types as supporting the buffer protocol.
+For runtime uses, the ``ABC.register`` API can be used to register
+buffer classes with ``typing_extensions.Buffer``. When
+``types.Buffer`` is available, ``typing_extensions`` should simply
+re-export it.
+
+
+No special meaning for ``bytes``
+--------------------------------
+
+The special case stating that ``bytes`` may be used as a shorthand
+for other ``ByteString`` types will be removed from the ``typing``
+documentation.
+With ``types.Buffer`` available as an alternative, there is no good
+reason to allow ``bytes`` as a shorthand.
+We suggest that type checkers currently implementing this behavior
+should deprecate and eventually remove it.
+
+
+Backwards Compatibility
+=======================
+
+As the runtime changes in this PEP only add a new class, there are
+no backwards compatibility concerns.
+
+However, the recommendation to remove the special behavior for
+``bytes`` in type checkers does have a backwards compatibility
+impact on their users. An `experiment <https://github.com/python/mypy/pull/12661>`__
+with mypy shows that several major open source projects that use it
+for type checking will see new errors if the ``bytes`` promotion
+is removed. However, many of these errors can be fixed by improving
+the stubs in typeshed, as already done for the
+`builtins <https://github.com/python/typeshed/pull/7631>`__,
+`binascii <https://github.com/python/typeshed/pull/7677>`__,
+`pickle <https://github.com/python/typeshed/pull/7678>`__, and
+`re <https://github.com/python/typeshed/pull/7679>`__ modules.
+Overall, the change improves type safety and makes the type system
+more consistent, so we believe the migration cost is worth it.
+
+
+Security Implications
+=====================
+
+None.
+
+
+How to Teach This
+=================
+
+We will add notes pointing to ``types.Buffer`` to appropriate places in the
```

Comment:
```suggestion
We will add notes pointing to ``types.Buffer`` in appropriate places in the
```

Reply:
This doesn't fully solve the problem, since now "in" is repeated twice rather than "to", which is still potentially confusing and hard to read. It seems like I can't make a suggestion here, but what about something like 

```rst
We will add notes pointing to ``types.Buffer`` in appropriate places throughout the
```